mixins/localize/localize.js

```diff
@@ -0,0 +1,244 @@
+import { getLocalizeOverrideResources } from '../../helpers/getLocalizeResources.js';
+import IntlMessageFormat from 'intl-messageformat';
+
+export const allowedTags = Object.freeze(['d2l-link', 'd2l-tooltip-help', 'p', 'br', 'b', 'strong', 'i', 'em', 'button']);
+
+const getDisallowedTagsRegex = allowedTags => {
+	const validTerminators = '([>\\s/]|$)';
+	const allowedAfterTriangleBracket = `/?(${allowedTags.join('|')})?${validTerminators}`;
+	return new RegExp(`<(?!${allowedAfterTriangleBracket})`);
+};
+
+export const disallowedTagsRegex = getDisallowedTagsRegex(allowedTags);
+const noAllowedTagsRegex = getDisallowedTagsRegex([]);
+
+export const getLocalizeClass = (superclass = class {}) => class LocalizeClass extends superclass {
+
+	static documentLocaleSettings = getDocumentLocaleSettings();
+	pristine = true;
+	#resourcesPromise;
+	#resolveResourcesLoaded;
+	localizeMarkup = localizeMarkup;
+
+	#localeChangeCallback;
+	async #defaultLocaleChangeCallback() {
+		if (!this._hasResources()) return;
+
+		const resourcesPromise = this.constructor._getAllLocalizeResources(this.config);
+		this.#resourcesPromise = resourcesPromise;
+		const localizeResources = (await resourcesPromise).flat(Infinity);
+		// If the locale changed while resources were being fetched, abort
+		if (this.#resourcesPromise !== resourcesPromise) return;
+
+		const allResources = {};
+		const resolvedLocales = new Set();
+		for (const { language, resources } of localizeResources) {
+			for (const [key, value] of Object.entries(resources)) {
+				allResources[key] = { language, value };
+				resolvedLocales.add(language);
+			}
+		}
+		this.localize.resources = allResources;
+		this.localize.resolvedLocales = resolvedLocales;
+		if (resolvedLocales.size > 1) {
+			console.warn(`Resolved multiple locales in '${this.constructor.name || this.tagName || ''}': ${[...resolvedLocales].join(', ')}`);
+		}
+
+		if (this.pristine) {
+			this.pristine = false;
+			this.#resolveResourcesLoaded();
+		}
+
+		this.#onResourcesChange();
+	}
+
+	#onResourcesChange() {
+		this.dispatchEvent?.(new CustomEvent('d2l-localize-resources-change'));
+		this.config?.onResourcesChange?.();
+		this.onLocalizeResourcesChange?.();
+	}
+
+	connect(cb = () => this.#defaultLocaleChangeCallback()) {
+		this.#localeChangeCallback = cb;
+		LocalizeClass.documentLocaleSettings.addChangeListener(this.#localeChangeCallback);
+		this.#localeChangeCallback();
+	}
+
+	disconnect() {
+		LocalizeClass.documentLocaleSettings.removeChangeListener(this.#localeChangeCallback);
+	}
+
+	localize(key) {
+
+		const { language, value } = this.localize.resources?.[key] ?? {};
+		if (!value) return '';
+
+		let params = {};
+		if (arguments.length > 1 && arguments[1]?.constructor === Object) {
+			// support for key-value replacements as a single arg
+			params = arguments[1];
+		} else {
+			// legacy support for localize-behavior replacements as many args
```

Comment:
Probably worth unit testing this, or if there's an existing test with the mixin moving that test to be with this code?